_posts/2013-01-05-(one-of)-london's-gdcr-2012.md

```diff
@@ -0,0 +1,256 @@
+---
+layout: post
+author: Gonçalo Silva
+authorname: Gonçalo Silva
+title: (one of) London's GDCR 2012
+preview: cross-posted from here
+---
+cross-posted from [here](http://www.gnorsilva.com/2013/01/global-day-of-code-retreat.html)
+
+Global day of code retreat took place worldwide in [154 locations](https://maps.google.com/maps/ms?msid=211858429594081017615.0004c84674c62aa900e06&msa=0) on Saturday 8th December. Over 36 hours, from New Zealand to Hawaii, almost 2000 developers came together to practice their craft.
+
+Me and [Tom Brand](https://twitter.com/tom_b025) facilitated one in London through the [London Software Craftsmanship Community](http://www.meetup.com/london-software-craftsmanship/events/88021082/). We were hosted by [Biomedcentral](http://www.biomedcentral.com/), and Tom's company [Lyagushka](http://lyagushka.co.uk/) kindly sponsored the catering.
+
+
+
+What is a code retreat? 
+---------------------------
+
+
+From [coderetreat.org](http://coderetreat.org/):
+
+"Coderetreat is a day-long, intensive practice event, focusing on the fundamentals of software development and design.
+
+
+By providing developers the opportunity to take part in focused practice, away from the pressures of 'getting things done', the coderetreat format has proven itself to be a highly effective means of skill improvement. 
+
+
+Practising the basic principles of modular and object-oriented design, developers can improve their ability to write code that minimizes the cost of change over time."
+
+
+
+How does it actually work?
+-----------------------------
+
+
+Developers come together to implement [Conway's Game of Life](http://en.wikipedia.org/wiki/Conway's_Game_of_Life) over 6 sessions of 45 minutes using any programming language they are comfortable with.
+
+Some sessions have challenging constraints, and at the end of each session the group gathers for a small retrospective.
+
+There are rules though:
+
+*   TDD or any other form of test first development is used
+
+*   Code written must be "simple":
+*   Passes all the tests.
+*   Maximises clarity with well expressed intent.
+*   Minimises duplication
+*   Has no superﬂuous parts.
+
+*   Pair programming (or 3 if there's an odd number) pairing with someone new every session
+
+*   At the end of each session **code is deleted**
+
+
+Deleting code might seem unnecessary or harsh but its important as the main point of this day is to practice and learn, not to finish. Its not a competition and developers should aim at creating the best code they can within the set constraint, and have fun!
+
+
+How did it go?
+-----------------
+
+
+Though we had over 20 people subscribed to attend only 9 showed up, but this actually worked out well as we could focus more on each pair.
+
+We started at 9 am with a quick intro about the LSCC, what a code retreat is all about, format of the day, and an explanation of Conway's Game of Life. We then jumped straight to the first session.
+
+
+###
```

Comment:
I have reviewed most of them and visited some post.
The only problem I could see is that the new line is breaking the style here: https://deploy-preview-279--coderetreat-org.netlify.com/blog/2013/01/05/(one-of)-london's-gdcr-2012/

Great job :grinning: